upup/pkg/fi/cloudup/awstasks/launchtemplate_target_api.go

```diff
@@ -301,68 +301,53 @@ func (t *LaunchTemplate) findAllLaunchTemplates(c *fi.Context) ([]*ec2.LaunchTem
 	}
 }
 
-// findAllLaunchTemplateVersions returns all the launch templates versions for us
-func (t *LaunchTemplate) findAllLaunchTemplatesVersions(c *fi.Context) ([]*ec2.LaunchTemplateVersion, error) {
-	var list []*ec2.LaunchTemplateVersion
-
+// findLaunchTemplates returns a list of launch templates
+func (t *LaunchTemplate) findLaunchTemplates(c *fi.Context) ([]*ec2.LaunchTemplateVersion, error) {
 	cloud, ok := c.Cloud.(awsup.AWSCloud)
 	if !ok {
 		return []*ec2.LaunchTemplateVersion{}, fmt.Errorf("invalid cloud provider: %v, expected: awsup.AWSCloud", c.Cloud)
 	}
 
+	// @step: get a list of the launch templates
 	templates, err := t.findAllLaunchTemplates(c)
 	if err != nil {
 		return nil, err
 	}
 
+	prefix := fmt.Sprintf("%s-", fi.StringValue(t.Name))
+
+	// @step: get the launch template versions for the templates we are interested in
+	var list []*ec2.LaunchTemplateVersion
 	var next *string
 	for _, x := range templates {
-		err := func() error {
-			for {
-				resp, err := cloud.EC2().DescribeLaunchTemplateVersions(&ec2.DescribeLaunchTemplateVersionsInput{
-					LaunchTemplateName: x.LaunchTemplateName,
-					NextToken:          next,
-				})
-				if err != nil {
-					return err
-				}
-				list = append(list, resp.LaunchTemplateVersions...)
-				if resp.NextToken == nil {
-					return nil
+		if strings.HasPrefix(aws.StringValue(x.LaunchTemplateName), prefix) {
```

Comment:
Totally out of scope for this PR, but apparently filters can also support wildcards, e.g. `fmt.Sprintf("%s-*", name)`.  I certainly didn't know that - so it wouldn't surprise me if we had multiple places where we could pre-filter by prefix.

Source: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Filtering.html#Filtering_Resources_CLI